recaf-ui/src/main/java/me/coley/recaf/ui/pane/assembler/suggestion/WorkspaceClassTree.java

```diff
@@ -0,0 +1,52 @@
+package me.coley.recaf.ui.pane.assembler.suggestion;
+
+import me.coley.recaf.ControllerListener;
+import me.coley.recaf.code.ClassInfo;
+import me.coley.recaf.util.ClasspathUtil;
+import me.coley.recaf.util.Streams;
+import me.coley.recaf.util.threading.ThreadUtil;
+import me.coley.recaf.workspace.Workspace;
+import me.coley.recaf.workspace.WorkspaceListener;
+import me.coley.recaf.workspace.resource.Resource;
+
+import java.util.concurrent.CompletableFuture;
+
+public class WorkspaceClassTree implements ControllerListener, WorkspaceListener {
+	private static CompletableFuture<ClasspathUtil.Tree> future; // TODO bad static
```

Comment:
Yes, but also if the class itself is not a tree, the class name ought to be changed.